test/integration/test_http.py

```diff
@@ -129,6 +129,54 @@ def test_basicTxn(self):
         status = txn.end()
         self.assertTrue(status.is2xxOK())
 
+        # Test partial update
+        # Begin Txn
+        status, txn = TestHTTP.cl.begin_txn()
+        self.assertTrue(status.is2xxOK())
+
+        # Write partial upate
+        record = TestHTTP.schema.make_record(data=b"mydata_update")
+        record.fieldsForPartialUpdate = [b"data"]
+        key = TestHTTP.schema.make_record(partitionKey=b"test1pk", rangeKey=b"test1rk")
+        status = txn.write(TestHTTP.cname, record, key=key)
+        self.assertTrue(status.is2xxOK(), msg=status.message)
+
+        # read data
+        status, resultRec = txn.read(TestHTTP.cname, key)
+        self.assertTrue(status.is2xxOK());
+        self.assertEqual(resultRec.fields.partitionKey, b"test1pk")
+        self.assertEqual(resultRec.fields.rangeKey, b"test1rk")
+        self.assertEqual(resultRec.fields.data, b"mydata_update")
+
+        # Commit
+        status = txn.end()
+        self.assertTrue(status.is2xxOK())
+
+        # Update range key using partial update
+        # Begin Txn
+        status, txn = TestHTTP.cl.begin_txn()
+        self.assertTrue(status.is2xxOK())
+
+        # Write partial upate
+        record = TestHTTP.schema.make_record(rangeKey=b"test1rk_update")
+        record.fieldsForPartialUpdate = [b"rangeKey"]
+        key = TestHTTP.schema.make_record(partitionKey=b"test1pk", rangeKey=b"test1rk")
+        status = txn.write(TestHTTP.cname, record, key=key)
+        self.assertTrue(status.is2xxOK())
+
+        # read using old range key, should fail but succeeds as data seems to be
+        # indexed by old range key but, it's value is new range key
```

Comment:
We can disallow this for the proxy as mentioned above but this is a good find since the k23si client should check for this. I will add an issue for it.

Reply:
Done.